### PR TITLE
DROOLS-4762: [DMN Designer] Data Types - Tab index is not working on Data Type row

### PR DESCRIPTION
**Pull request by danielzhe**

### CODE REVIEW

**Comment by jomarko** [comment]
@danielzhe it seems we have a related failing unit test, could you have a look on it please?

**Comment by jomarko** [comment]
### Manual review
The tab control was improved rapidly. Great job. Just two questions.

1. The tab control stop working after constraint editor was invoked, see the [video](https://issues.jboss.org/secure/attachment/12458072/constraints-focus.webm), should I report this?
2. The tab control works differently for sub fields of structure and top level entries, see the [video](https://issues.jboss.org/secure/attachment/12458073/edit-focus.webm), Firstly I navigate there to edit button of sub-entry and press enter - nothing happens, secondly I navigate to edit button of top level entry, press enter-  the edit mode is invoked. Should I report this?

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/types/listview/DataTypeListItem.java [hunk 1]
```diff
@@ -218,14 +218,17 @@ public void collapse() {
         view.collapse();
     }
 
-    void refreshSubItems(final List<DataType> dataTypes) {
+    void refreshSubItems(final List<DataType> dataTypes, boolean enableFocusMode) {
```
[the remark this answers is not in the document]
Comment (danielzhe) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/types/listview/DataTypeListItem.java]: Sure.

**Comment by danielzhe** [comment]
@jomarko Thank you for your review! In my opinion, the 1 is a real issue that should be reported. We have more work to to do improve keyboard navigation.
About the issue 2, in the video you was editing "tCar" and tried to start editing "Color" without finishing to edit "tCar". I noticed that even if you click on the edit button fro "Color" when you are editing "tCar", nothing happens. I think this is another issue that we have do address. I think we should discard the changes in "tCar" in this case and start to editing the "Color". Maybe this can be addressed in this PR too if we decide how that can be handled.

**Pull request merged**